src/Event/Http/HttpRequestEvent.php

```diff
@@ -125,25 +138,40 @@ public function getRequestContext(): array
 
     public function getCookies(): array
     {
-        if (! isset($this->headers['cookie'])) {
-            return [];
-        }
+        $cookies = [];
+        if ($this->getPayloadVersion() >= 2 && isset($this->event['cookies'])) {
+            $cookieParts = $this->event['cookies'];
+        } else {
+            if (! isset($this->headers['cookie'])) {
+                return [];
+            }
 
-        // Multiple "Cookie" headers are not authorized
-        // https://stackoverflow.com/questions/16305814/are-multiple-cookie-headers-allowed-in-an-http-request
-        $cookieHeader = $this->headers['cookie'][0];
+            // Multiple "Cookie" headers are not authorized
+            // https://stackoverflow.com/questions/16305814/are-multiple-cookie-headers-allowed-in-an-http-request
+            $cookieHeader = $this->headers['cookie'][0];
+
+            $cookieParts = explode('; ', $cookieHeader);
+        }
 
-        $cookies = [];
-        $cookieParts = explode('; ', $cookieHeader);
         foreach ($cookieParts as $cookiePart) {
             [$cookieName, $cookieValue] = explode('=', $cookiePart, 2);
             $cookies[$cookieName] = urldecode($cookieValue);
         }
+
         return $cookies;
     }
 
+    public function getPayloadVersion(): ?float
+    {
+        return $this->payloadVersion;
+    }
+
     private function rebuildQueryString(): string
     {
+        if (isset($this->event['rawQueryString'])) {
+            return $this->event['rawQueryString'];
+        }
```

Comment:
Same as with `rawPath` -> we should differentiate entirely between v1 and v2 events. That will ease readability

```suggestion
        if ($this->payloadVersion >= 2) {
            return $this->event['rawQueryString'];
        }
```